src/main/java/org/extism/chicory/sdk/ChicoryModule.java

```diff
@@ -0,0 +1,46 @@
+package org.extism.chicory.sdk;
+
+import com.dylibso.chicory.aot.AotMachine;
+import com.dylibso.chicory.runtime.Instance;
+import com.dylibso.chicory.wasm.Module;
+import com.dylibso.chicory.wasm.Parser;
+
+import java.nio.file.Path;
+
+public class ChicoryModule {
+
+    static final boolean IS_NATIVE_IMAGE_AOT = Boolean.getBoolean("com.oracle.graalvm.isaot");
```

Comment:
this is not strictly related to this PR, but it's necessary to compile this under graalvm native image. This flag is true when the native image builder it's running. We want to make sure that the code path below is pruned at build-time, because the AoT in its current state is not compatible with the way native image works.